zipkin/zipkin_data.py

```diff
@@ -40,6 +40,57 @@ def from_hex(cls, s):
         return cls(cls.STRUCT.unpack(binascii.unhexlify(s))[0])
 
 
+class ZipkinTraceId(object):
+    '''ZipkinTraceId repesents as 128 bit Zipkin Trace ID made up of 2 x 64-bit
+    signed numbers'''
+
+    MAX_VAL = 2 ** 63 - 1
+    MIN_VAL = -MAX_VAL
+
+    STRUCT = struct.Struct('!qq')
+
+    def __init__(self, low, high):
+        if low < self.MIN_VAL or low > self.MAX_VAL:
+            raise ValueError(
+                "low bytes %d are not in the allowed range of [%d, %d]" %
+                (low, self.MIN_VAL, self.MAX_VAL))
+        if high < self.MIN_VAL or high > self.MAX_VAL:
+            raise ValueError(
+                "high bytes %d are not in the allowed range of [%d, %d]"
+                % (high, self.MIN_VAL, self.MAX_VAL))
+
+        self.low = low
+        self.high = high
+
+    def get_binary_low_bytes(self):
+        '''Return the low 64bit number of the Trace ID. Originally Trace IDs
+        were 64bits and thus this number would represent the entire Trace ID'''
+        return self.low
+
+    def get_binary_high_bytes(self):
+        '''Return the high 64bit number of the Trace ID, if set. Originally,
```

Comment:
Here you say "if set", is there a path that this can be None, or is this to cover 0 bytes? (the classmethods below seem to prevent construction with None values).

Reply:
It's just to cover the `0` case (e.g. if upstream is still using 64bit trace ids). However, it's a bit moot because it's indistinguishable from having set these bits to zeros. I'll amend the comment